lib/util.js

```diff
@@ -18,6 +20,9 @@ const files1 = fs.readdirSync(`${__dirname}/profiles/TR/`),
  */
 
 const buildJSONresult = function(err, warn, inf, metadata) {
+    err.every(insertDetailMessage);
```

Comment:
The issue with adding the `detailMessage` in `buildJSONresult` is that it's only available when pubrules is done processing all the rules (see https://github.com/w3c/specberus/blob/main/lib/validator.js#L69). It would make more sense to add the message on each error/warning/info detected (on validator.js from lines [L.48](https://github.com/w3c/specberus/blob/main/lib/validator.js#L48) to [L.50](https://github.com/w3c/specberus/blob/main/lib/validator.js#L50) and [L.106](https://github.com/w3c/specberus/blob/main/lib/validator.js#L106) to [L.108](https://github.com/w3c/specberus/blob/main/lib/validator.js#L108)

Reply:
I add the message in [validator.js](https://github.com/w3c/specberus/pull/999/files#diff-75a677c2b40d5af505ce7b663015040bc3ff16953b75b43e44ddebe88c59d836) for `Specberus.prototype.error`, when emitting the error/warning.

```
this.sink.emit('warning', rule, {
    key: key,
    extra: extra,
    detailMessage: l10n.assembleData(null, rule, key, extra).message
});
```
So in https://github.com/w3c/specberus/blob/main/lib/validator.js#L107 you'll have it in `data` already.
```
self.sink.on('warning', function (...data) { warnings.push(Object.assign({}, ...data)); });
```